src/game/AI/BaseAI/GuardAI.cpp

```diff
@@ -138,6 +138,6 @@ void GuardAI::AttackStart(Unit* u)
 
 void GuardAI::JustDied(Unit* killer)
 {
-    if (Player* pkiller = killer->GetCharmerOrOwnerPlayerOrPlayerItself())
+    if (Player* pkiller = killer->GetBeneficiaryPlayer())
```

Comment:
@cyberium I had the same argument, but he is right, thats what it essentially does and I wasn't able to find a reason against that name, believe me, I tried ^^